pandas/core/frame.py

```diff
@@ -1520,6 +1520,28 @@ def to_feather(self, fname):
         from pandas.io.feather_format import to_feather
         to_feather(self, fname)
 
+    def to_parquet(self, fname, engine, compression=None,
+                   **kwargs):
+        """
+        write out the binary parquet for DataFrames
+
+        .. versionadded:: 0.20.0
+
+        Parameters
+        ----------
+        fname : str
+            string file path
```

Comment:
we handle path_or_buffers in the following way:

- path-like objects (``pathlib`` and ``py.local``), we stringify
- if a string
  - if its a url we turn this into a Bytes object (also handling gzip content encoding) 
  - if its a s3 url we defer to ``s3fs`` for opening
  - else we would do things like expand_user
- file-like we pass thru
- for csv reading we will handle the file io & encoding
- all others we pass the string-path thru

So i don't see any reason to handle this differently. The IO engine gets to handle a fully qualified string path. (e.g. HDF5, excel, pickle, json) look all the same to pandas. The IO engine is in charge of opening closing the actual files.